ui/components/ApolloProvider/index.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { ReactNode } from 'react'
+import { ApolloClient, HttpLink, InMemoryCache, NormalizedCacheObject } from 'apollo-boost'
+import { ApolloProvider as ApolloHookProvider } from 'react-apollo-hooks'
+import { useConfig } from '../ConfigProvider'
+
+interface ApolloProviderProps {
+  children: ReactNode
+  client?: ApolloClient<NormalizedCacheObject>
+}
+
+export function ApolloProvider(props: ApolloProviderProps) {
```

Comment:
I preferred the old pattern of initializing the Apollo client in the appropriate root files for the server and client, to be honest. It's more clear what's happening, and doesn't require a custom provider component.

Imagine being a future person reading this code. When does a `client` get passed in? Can I trust that the initialization here is the one I should be looking at the debug my issue? I have to chase down that `client` prop and figure out where and when it's set. It's more complex, harder to read.

Always keep in mind, code is read more than it's written. You should optimize it for the reader.

Reply:
It's just that somehow I need to get the url from the context provider using a hook. And the rules for hooks make doing this any other way nearly impossible

Reply:
You have access to the config in `public/ui.tsx`, just not from context. That should be fine.